doc/warm-reboot/view_switch.md

```diff
@@ -0,0 +1,179 @@
+# View Switching in ProducerStateTable
+
+### Problem Description
+To support warm reboot, 
+ProducerStateTable needs to support a feature that producer can create a temporary view for a table, 
+into which write operation will not not be synced to consumers immediately.
+Instead, the producer can explicitly ask to apply the view, 
+upon which all the objects in new view will get synced to consumer.
+
+However, the actual operations get passed to consumer should be optimized as best effort, 
+so if an object is present in both old view and new view, 
+it should not be `del()` and `set()` again.
+
+In order to archieve that, we are going to add two additional APIs: `create_temp_view()` and `apply_temp_view()`,
+as well as modifying existing `del()` and `set()` API.
+
+This design is based on the assumption that there is only one producer writing to a specific table. 
+If there are multiple producer writing to a same table, the changes that 'secondary' producer write to the table
+while 'main' producer was doing view switching might be lost.
```

Comment:
We probably only have one producer for a table for now. But is there any mechanism to enforce this constrain? or make some of the steps atomic to database?